File: AI聊天室/src/utils/schematicReviewHistory.ts
import axios from 'axios';
import { apiUrl } from '@/utils/apiBase';
import type { SchematicCheckDispositionMap } from '@/utils/schematicReview';
import type { AIReviewEntry } from '@/components/NetlistResultsPanel';

export interface SchematicReviewHistorySummary {
  id: string;
  title: string;
  netlist_result_id?: string | null;
  summary_pass: number;
  summary_warning: number;
  summary_info: number;
  created_at: string;
}

export interface SchematicReviewHistoryPayload {
  aggregated_review_summary: any;
  ai_review_entries: Array<{
    id: string;
    content: string;
    parsed: any;
    timestamp?: string;
    aiName: string;
  }>;
  cleaned_netlist_text: string;
  check_dispositions: SchematicCheckDispositionMap;
  default_ai_name: string;
  netlist_name: string;
}

export interface SchematicReviewHistoryRecord extends SchematicReviewHistorySummary {
  payload: SchematicReviewHistoryPayload;
}

export async function listSchematicReviewHistory(): Promise<SchematicReviewHistorySummary[]> {
  const res = await axios.get(apiUrl('/api/schematic-review/history'), { withCredentials: true });
  if (!res.data?.success) throw new Error(res.data?.error || '加载历史失败');
  return res.data.records || [];
}

export async function getSchematicReviewHistory(
  id: string
): Promise<SchematicReviewHistoryRecord> {
  const res = await axios.get(apiUrl(`/api/schematic-review/history/${id}`), {
    withCredentials: true,
  });
  if (!res.data?.success) throw new Error(res.data?.error || '加载记录失败');
  return res.data.record;
}

export async function saveSchematicReviewHistory(body: {
  title: string;
  netlist_name: string;
  netlist_result_id: string | null;
  summary: { pass: number; warning: number; info: number };
  aggregated_review_summary: any;
  ai_review_entries: AIReviewEntry[];
  cleaned_netlist_text: string;
  check_dispositions: SchematicCheckDispositionMap;
  default_ai_name: string;
}): Promise<SchematicReviewHistorySummary> {
  const entries = body.ai_review_entries.map((e) => ({
    id: e.id,
    content: e.content,
    parsed: e.parsed,
    timestamp: e.timestamp instanceof Date ? e.timestamp.toISOString() : e.timestamp,
    aiName: e.aiName,
  }));
  const res = await axios.post(
    apiUrl('/api/schematic-review/history'),
    { ...body, ai_review_entries: entries },
    { withCredentials: true }
  );
  if (!res.data?.success) throw new Error(res.data?.error || '保存历史失败');
  return res.data.record;
}

export function parseHistoryAiReviewEntries(
  entries: SchematicReviewHistoryPayload['ai_review_entries']
): AIReviewEntry[] {
  return (entries || []).map((e) => ({
    id: e.id,
    content: e.content,
    parsed: e.parsed,
    timestamp: e.timestamp ? new Date(e.timestamp) : new Date(),
    aiName: e.aiName,
  }));
}

File: AI聊天室/backend/services/schematic_review_history.py
"""原理图 AI 审核 — 用户评审历史记录"""

from __future__ import annotations

from typing import Any, Dict, List, Optional


def serialize_review_payload(
    *,
    aggregated_review_summary: Any,
    ai_review_entries: List[Dict],
    cleaned_netlist_text: str,
    check_dispositions: Dict,
    default_ai_name: str,
    netlist_name: str,
) -> Dict[str, Any]:
    entries = []
    for e in ai_review_entries or []:
        ts = e.get("timestamp")
        if hasattr(ts, "isoformat"):
            ts_str = ts.isoformat()
        else:
            ts_str = str(ts) if ts else None
        entries.append(
            {
                "id": e.get("id"),
                "content": e.get("content"),
                "parsed": e.get("parsed"),
                "timestamp": ts_str,
                "aiName": e.get("aiName"),
            }
        )
    return {
        "aggregated_review_summary": aggregated_review_summary,
        "ai_review_entries": entries,
        "cleaned_netlist_text": cleaned_netlist_text or "",
        "check_dispositions": check_dispositions or {},
        "default_ai_name": default_ai_name or "",
        "netlist_name": netlist_name or "",
    }


def normalize_loaded_payload(payload: Dict[str, Any]) -> Dict[str, Any]:
    entries = payload.get("ai_review_entries") or []
    normalized_entries = []
    for e in entries:
        ts_raw = e.get("timestamp")
        normalized_entries.append(
            {
                **e,
                "timestamp": ts_raw,
            }
        )
    return {
        **payload,
        "ai_review_entries": normalized_entries,
    }
